packages/keystone/adapterCache.js

```diff
@@ -0,0 +1,425 @@
+                connectedLists[relationName] = listName
+            }
+        }
+    }
+
+    logger.info({ connectedLists })
+
+    // Step 2: Iterate over lists, patch mutations and queries inside list.
+
+    for (const listAdapter of listAdapters) {
+
+        const listName = listAdapter.key
+        cache[listName] = {}
+
+        const fields = listAdapter.fieldAdaptersByPath
+        if (!fields[UPDATED_AT_FIELD] || !fields[UPDATED_AT_FIELD]) {
+            logger.info(`ADAPTER_CACHE: Cache is NOT enabled for list: ${listName} -> No ${UPDATED_AT_FIELD} field`)
+            continue
+        }
+
+        if (excludedLists.includes(listName) ||
+            !includeAllLists && !includedLists.includes(listName)) {
+            logger.info(`ADAPTER_CACHE: Cache is NOT enabled for list: ${listName} -> Cache is not included by config`)
+            continue
+        }
+
+        logger.info(`ADAPTER_CACHE: Cache is enabled for list: ${listName}`)
+
+        // Patch public queries from BaseKeystoneList:
+
+        const originalItemsQuery = listAdapter.itemsQuery
+        const getItemsQueryKey = ([args, opts]) => `${JSON.stringify(args)}_${get(opts, 'from', null)}_${JSON.stringify(get(opts, ['context', 'authedItem', 'id']))}`
+        const getItemsQueryCondition = ([args]) => get(args, ['where'])
+        listAdapter.itemsQuery = patchAdapterQueryFunction(listName, 'itemsQuery', originalItemsQuery, listAdapter, middleware, getItemsQueryKey, getItemsQueryCondition, relations)
+
+        const originalFind = listAdapter.find
+        const getFindKey = ([condition]) => `${JSON.stringify(condition)}`
+        const getFindCondition = ([condition]) => condition
+        listAdapter.find = patchAdapterQueryFunction(listName, 'find', originalFind, listAdapter, middleware, getFindKey, getFindCondition, relations)
+
+        const originalFindById = listAdapter.findById
+        const getFindByIdKey = ([id]) => `${id}`
+        listAdapter.findById = patchAdapterQueryFunction(listName, 'findById', originalFindById, listAdapter, middleware, getFindByIdKey)
+
+        const originalFindOne = listAdapter.findOne
+        const getFindOneKey = ([condition]) => `${JSON.stringify(condition)}`
+        listAdapter.findOne = patchAdapterQueryFunction(listName, 'findOne', originalFindOne, listAdapter, middleware, getFindOneKey, getFindCondition, relations)
+
+        // Patch mutations:
+
+        const originalUpdate = listAdapter.update
+        listAdapter.update = patchAdapterFunction(listName, 'update', originalUpdate, listAdapter, middleware, connectedLists )
+
+        const originalCreate = listAdapter.create
+        listAdapter.create = patchAdapterFunction(listName, 'create', originalCreate, listAdapter, middleware, connectedLists )
+
+        const originalDelete = listAdapter.delete
+        listAdapter.delete = patchAdapterFunction(listName, 'delete', originalDelete, listAdapter, middleware, connectedLists )
```

Comment:
Many need to patch too?

Reply:
It seems like keystone uses only create function from adapter ... please check out this code...
<img width="606" alt="image" src="https://user-images.githubusercontent.com/33755274/225427102-bf21ef06-2449-45f6-8657-1c5074e22252.png">

node_modules/@keystonejs/keystone/lib/ListTypes/list.js 715 -> 728